File: usvm-python/usvm-python-main/src/main/java/org/usvm/interpreter/MemberDescriptor.java
package org.usvm.interpreter;

import org.jetbrains.annotations.Nullable;
import org.usvm.language.SymbolForCPython;
import org.usvm.machine.symbolicobjects.UninterpretedSymbolicPythonObject;

public abstract class MemberDescriptor {
    @Nullable
    public abstract SymbolForCPython getMember(ConcolicRunContext ctx, @Nullable UninterpretedSymbolicPythonObject owner);
}

File: usvm-python/usvm-python-main/src/main/kotlin/org/usvm/machine/interpreters/symbolic/operations/descriptors/MemberDescriptor.kt
package org.usvm.machine.interpreters.symbolic.operations.descriptors

import org.usvm.machine.ConcolicRunContext
import org.usvm.machine.interpreters.concrete.utils.SymbolForCPython
import org.usvm.machine.symbolicobjects.UninterpretedSymbolicPythonObject

interface MemberDescriptor {
    fun getMember(ctx: ConcolicRunContext, owner: UninterpretedSymbolicPythonObject?): SymbolForCPython?
}
